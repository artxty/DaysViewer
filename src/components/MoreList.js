import React from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  Text,
  View,
  SectionList,
} from 'react-native';
import iosColors from 'ios-colors';
// import Constants from 'expo-constants';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // paddingTop: 30,
    backgroundColor: iosColors.systemGray6.light,
  },
  header: {
    paddingHorizontal: 16,
    fontSize: 13,
    color: iosColors.systemGray2.dark,
    marginBottom: 5,
  },
  item: {
    backgroundColor: '#fff',
    paddingVertical: 10,
    paddingHorizontal: 15,
  },
  itemSeparator: {
    borderColor: iosColors.systemGray3.light,
    borderBottomWidth: 0.5,
  },
  sectionSeparator: {
    borderColor: iosColors.systemGray3.light,
    borderBottomWidth: 0.5,
  },
  renderSectionFooter: {
    marginBottom: 30,
  },
});

export default function MoreList({ sections }) {
  return (
    <View style={styles.container}>
      <SectionList
        ListHeaderComponent={() => <View style={styles.renderSectionFooter} />}
        // ListFooterComponent={() => <Text>{`Calendar App v${Constants.nativeAppVersion}`}</Text>}
        sections={sections}
        keyExtractor={(item, index) => item + index}
        renderItem={({ item }) => <View style={styles.item}>{item}</View>}
        renderSectionHeader={
          ({ section: { title } }) => <Text style={styles.header}>{title.toUpperCase()}</Text>
        }
        stickySectionHeadersEnabled={false}
        ItemSeparatorComponent={() => <View style={styles.itemSeparator} />}
        SectionSeparatorComponent={() => <View style={styles.sectionSeparator} />}
        renderSectionFooter={() => <View style={styles.renderSectionFooter} />}
      />
    </View>
  );
}

MoreList.propTypes = {
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
};
