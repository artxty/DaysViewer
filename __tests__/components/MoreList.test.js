import React from 'react';
import renderer from 'react-test-renderer';

import MoreList from '../../src/components/MoreList';

jest.mock('expo-constants', () => ({ nativeAppVersion: 'v0.1.1' }));
jest.mock('react-native', () => {
  const reactNative = jest.requireActual('react-native');
  return {
    __esModule: true,
    View: 'View',
    Text: 'Text',
    SectionList: 'SectionList',
    StyleSheet: reactNative.StyleSheet,
  };
});

describe('MoreList', () => {
  it('renders correctly', () => {
    const sections = [
      { titile: '1', data: [1, 2, 3] },
      { titile: '2', data: [3, 4, 5] },
    ];
    const tree = renderer.create(<MoreList sections={sections} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
