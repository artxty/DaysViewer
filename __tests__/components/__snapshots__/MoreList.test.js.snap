// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreList renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 247)",
      "flex": 1,
    }
  }
>
  <SectionList
    ItemSeparatorComponent={[Function]}
    ListHeaderComponent={[Function]}
    SectionSeparatorComponent={[Function]}
    keyExtractor={[Function]}
    renderItem={[Function]}
    renderSectionFooter={[Function]}
    renderSectionHeader={[Function]}
    sections={
      Array [
        Object {
          "data": Array [
            1,
            2,
            3,
          ],
          "titile": "1",
        },
        Object {
          "data": Array [
            3,
            4,
            5,
          ],
          "titile": "2",
        },
      ]
    }
    stickySectionHeadersEnabled={false}
  />
</View>
`;
